src/pylexibank/dataset.py

```diff
@@ -361,6 +407,194 @@ def coverage(self, vars, glangs, c):  # pragma: no cover
+
+        if self.metadata.related:
+            lines.extend(['See also %s' % self.metadata.related, ''])
+
+        if self.metadata.conceptlist:
+            lines.extend([
+                'Conceptlist in Concepticon: [{0}](http://concepticon.clld.org/contributions'
+                '/{0})'.format(self.metadata.conceptlist),
+                ''])
+
+        # add NOTES.md
+        if self.dir.joinpath('NOTES.md').exists():
+            lines.extend(['## Notes', ''])
+            lines.extend(self.dir.joinpath('NOTES.md').read_text().split("\n"))
+            lines.extend(['', ''])  # some blank lines
+
+        synonyms = defaultdict(Counter)
+        totals = {
+            'languages': Counter(),
+            'concepts': Counter(),
+            'sources': Counter(),
+            'cognate_sets': Counter(),
+            'lexemes': 0,
+        }
+
+        missing_source = []
+        missing_lang = [L['NAME'] for L in self.languages if not L['GLOTTOCODE']]
+        missing_param = []
+        missing_concept = [c for c in self.concepts if not c['CONCEPTICON_ID']]
+
+        for row in self.cldf['FormTable']:
+            if row['Source']:
+                totals['sources'].update(['y'])
+            else:
+                missing_source.append(row)
+            if row['Parameter_ID']:
+                totals['concepts'].update([row['Parameter_ID']])
+            else:
+                missing_param.append(row)
+            totals['languages'].update([row['Language_ID']])
+            totals['lexemes'] += 1
+            if row['Language_ID'] and row['Parameter_ID']:
+                synonyms[row['Language_ID']].update([row['Parameter_ID']])
+
+        for row in self.cldf['CognateTable']:
+            totals['cognate_sets'].update([row['Cognateset_ID']])
+
+        sindex = sum(
+            [sum(list(counts.values())) / float(len(counts)) for counts in synonyms.values()])
+        langs = set(synonyms.keys())
+        if langs:
+            sindex /= float(len(langs))
+        else:
+            sindex = 0
+        totals['SI'] = sindex
+
+        stats = tr_analysis['stats']
+        lsegments = len(stats['segments'])
+        lbipapyerr = len(stats['bipa_errors'])
+        lsclasserr = len(stats['sclass_errors'])
+
+        def ratio(prop):
+            return sum(v for k, v in totals[prop].items() if k) / float(totals['lexemes'])
+
+        badges = [
+            self.build_status_badge(),
+            get_badge(ratio('languages'), 'Glottolog'),
+            get_badge(ratio('concepts'), 'Concepticon'),
+            get_badge(ratio('sources'), 'Source'),
+        ]
+        if lsegments:
+            badges.extend([
+                get_badge((lsegments - lbipapyerr) / lsegments, 'BIPA'),
+                get_badge((lsegments - lsclasserr) / lsegments, 'CLTS SoundClass'),
+            ])
+        lines.extend(['## Statistics', '\n', '\n'.join(badges), ''])
+        stats_lines = [
```

Comment:
Could the following be put into a separate template file? Users might find it nice to create customised reports (not that it would be hard to manually modify the textual output, just an idea).

Reply:
Whops, should have continued reading. Sorry!